Tests/XcresultparserTests/XcresultparserTests.swift

```diff
@@ -101,50 +101,61 @@ Summary
     }
 
     func testCoverageConverter() throws {
-        let xcresultFile = Bundle.module.path(forResource: "test", ofType: "xcresult")!
+        let xcresultFile = Bundle.module.url(forResource: "test", withExtension: "xcresult")!
         let projectRoot = ""
         let quiet = 1
 
-        guard let converter = SonarCoverageConverter(with: URL(fileURLWithPath: xcresultFile), projectRoot: projectRoot) else {
+        guard let converter = SonarCoverageConverter(with: xcresultFile,
+                                                     projectRoot: projectRoot) else {
             XCTFail("Unable to create CoverageConverter from \(xcresultFile)")
             return
         }
         let rslt = try converter.xmlString(quiet: quiet == 1)
         XCTAssertTrue(rslt.starts(with: "<coverage version=\"1\">"))
+
+        // Runs are non deterministic a simple compare is not doable for tests
+        //assertXmlTestReportsAreEqual(expectedFileName: "sonarCoverage", actual: converter)
+    }
+
+    func testCoberturaConverter() throws {
+        let xcresultFile = Bundle.module.url(forResource: "test", withExtension: "xcresult")!
+        let projectRoot = ""
+        
+        guard let converter = CoberturaCoverageConverter(with: xcresultFile,
+                                                     projectRoot: projectRoot) else {
+            XCTFail("Unable to create CoverageConverter from \(xcresultFile)")
+            return
+        }
+        
+        assertXmlTestReportsAreEqual(expectedFileName: "cobertura", actual: converter)
     }
 
     func testJunitXMLSonar() throws {
-        let xcresultFile = Bundle.module.path(forResource: "test", ofType: "xcresult")!
+        let xcresultFile = Bundle.module.url(forResource: "test", withExtension: "xcresult")!
         let projectRoot = ""
         guard let junitXML = JunitXML(
-            with: URL(fileURLWithPath: xcresultFile),
+            with: xcresultFile,
             projectRoot: projectRoot,
             format: .sonar
         ) else {
             XCTFail("Unable to create JunitXML from \(xcresultFile)")
             return
         }
-
-        XCTAssertTrue(junitXML.xmlString.starts(with: "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"))
-        let dataXML = try! XMLDocument.init(xmlString: junitXML.xmlString, options: [])
-        XCTAssertEqual(dataXML.rootElement()?.name, "testExecutions")
+        assertXmlTestReportsAreEqual(expectedFileName: "sonarTestExecution", actual: junitXML)
```

Comment:
👍
This is way better than just checking if the rootElement's name is testExecutions. Great idea to simply add the xml as asset and compare it.